listenstore/listenstore/listenstore.py

```diff
@@ -89,30 +87,48 @@ def insert_postgresql(self, listens):
                 if not listen.validate():
                     raise ValueError("Invalid listen: %s" % listen)
                 try:
-                    res = connection.execute(
-                    """ INSERT INTO listen(user_id, ts, artist_msid, album_msid, recording_msid, raw_data)
-                        VALUES ( %(user_id)s, to_timestamp(%(ts)s), %(artist_msid)s, %(album_msid)s,
-                        %(recording_msid)s, %(raw_data)s) ON CONFLICT DO NOTHING """, self.format_dict(listen))
+                    params = self.format_dict(listen)
+                    res = connection.execute(text("""
+                        INSERT INTO listen(user_id, ts, artist_msid, album_msid, recording_msid)
+                             VALUES (:user_id, to_timestamp(:ts), :artist_msid, :album_msid,
+                                    :recording_msid)
+                        ON CONFLICT DO NOTHING
+                          RETURNING id
+                    """), params)
+                    params['_id'] = res.fetchone()[0]
+
+                    res = connection.execute(text("""
+                        INSERT INTO listen_json(id, data)
+                             VALUES (:_id, :data)
+                        ON CONFLICT DO NOTHING
+                    """), params)
                 except Exception, e:     # Log errors
-                        self.log.error(e)
-
-    def execute(self, query, params={}):
-        with self.engine.connect() as connection:
-            res = connection.execute(query, params)
-            return res.fetchall()
+                    self.log.error(e)
 
     def fetch_listens_from_storage(self, user_id, from_id, to_id, limit, order, precision):
-        query = """ SELECT id, user_id, extract(epoch from ts), artist_msid, album_msid, recording_msid, raw_data """ + \
-                """ FROM listen WHERE user_id = %(user_id)s """ + \
-                """ AND extract(epoch from ts) > %(from_id)s AND extract(epoch from ts) < %(to_id)s  """ + \
-                """ ORDER BY extract(epoch from ts) """ + ORDER_TEXT[order] + """ LIMIT %(limit)s"""
-        params = {
-            'user_id': user_id,
-            'from_id': from_id,
-            'to_id': to_id,
-            'limit': limit
-        }
-
-        results = self.execute(query, params)
-        for row in results:
-            yield self.convert_row(row)
+        with self.engine.connect() as connection:
+            results = connection.execute(text("""
+                SELECT listen.id
+                     , user_id
+                     , extract(epoch from ts)
+                     , artist_msid
+                     , album_msid
+                     , recording_msid
+                     , data
+                  FROM listen
+                     , listen_json
+                 WHERE listen.id = listen_json.id
+                   AND user_id = :user_id
+                   AND extract(epoch from ts) > :from_id
```

Comment:
- I was not aware that you can pass the datetime objects the the query string. I thought only strings are allowed. Committed the changes. (I have used datetime objects on comparison but still i am using `extract(epoch from ts)` while returning the timestamps. The rest of the changes that make listenstore return datetime objects instead of epoch's is already written in [this](https://github.com/metabrainz/listenbrainz-server/pull/89) PR.)
- This PR was meant to be small and simle splitting of the listen table so that i can re-use the code in the api_compat branch. Making such changes was just not the purpose of this PR as i have already made of lot of changes in the api_compat branch. This also has some problems.   
 
  For examle, Consider the above change. I have made changes in the same part of code in api_compat branch, now i also submitted a PR with other changes in show-local-time-to-user branch and finally this PR which also does substantial changes in the same part of code. I now have 3 branches that are open and their changes are dependent on each other, making things difficult for me to manage. So i always aim at making one branch for formatting (for now it is api_compat) while the other branches that make the ground for the updates which i then use in the api_compat. I rebase the branch with those changes as they merge and continue working.
  
Please lets just stick to the main purpose of this PR for now. I'll appreciate your feedback on formatting on the api_compat branch.